src/m1_facilitator/SeedDataInitializer.ts

```diff
@@ -0,0 +1,197 @@
+   *
+   * @param originWeb3 Instance of origin web3.
+   * @param auxiliaryWeb3 Instance of auxiliary web3.
+   * @param gatewayAddresses Gateway address.
+   */
+  public async initialize(
+    originWeb3: Web3,
+    auxiliaryWeb3: Web3,
+    gatewayAddresses: string,
+  ): Promise<void> {
+    const erc20Gateway = Mosaic.interacts.getERC20Gateway(originWeb3, gatewayAddresses);
+    const cogatewayAddress = await erc20Gateway.methods.messageOutbox().call();
+    const erc20Cogateway = Mosaic.interacts.getERC20Cogateway(
+      auxiliaryWeb3,
+      cogatewayAddress,
+    );
+
+    const originAnchorAddress = await erc20Gateway.methods.stateRootProvider().call();
+    const originAnchorInstance = Mosaic.interacts.getAnchor(
+      originWeb3,
+      originAnchorAddress,
+    );
+
+    const auxiliaryAnchorAddress = await erc20Cogateway.methods.stateRootProvider().call();
+    const auxiliaryAnchorInstance = Mosaic.interacts.getAnchor(
+      auxiliaryWeb3,
+      auxiliaryAnchorAddress,
+    );
+
+    const originLatestAnchoredStateRootBlockHeight = await originAnchorInstance.methods
+      .getLatestStateRootBlockNumber().call();
+    const auxiliaryLatestAnchoredStateRootBlockHeight = await auxiliaryAnchorInstance.methods
+      .getLatestStateRootBlockNumber().call();
+
+    const originGateway = new Gateway(
+      Gateway.getGlobalAddress(gatewayAddresses),
+      Gateway.getGlobalAddress(cogatewayAddress),
+      GatewayType.ERC20,
+      Anchor.getGlobalAddress(originAnchorAddress),
+      new BigNumber(0),
+    );
+
+    const auxiliaryGateway = new Gateway(
+      Gateway.getGlobalAddress(cogatewayAddress),
+      Gateway.getGlobalAddress(gatewayAddresses),
+      GatewayType.ERC20,
+      Anchor.getGlobalAddress(auxiliaryAnchorAddress),
+      new BigNumber(0),
+    );
+
+    const originAnchor = new Anchor(
+      Anchor.getGlobalAddress(originAnchorAddress),
+      new BigNumber(originLatestAnchoredStateRootBlockHeight),
+    );
+
+    const auxiliaryAnchor = new Anchor(
+      Anchor.getGlobalAddress(auxiliaryAnchorAddress),
+      new BigNumber(auxiliaryLatestAnchoredStateRootBlockHeight),
+    );
+
+    const currentTimeStamp = Utils.getCurrentTimestamp();
+
+
+    const contractEntities = SeedDataInitializer.getContractEntities(
+      gatewayAddresses,
+      currentTimeStamp,
+      cogatewayAddress,
+      originAnchorAddress,
+      auxiliaryAnchorAddress,
+    );
+
+    const saveContractEntityPromises = contractEntities.map(
+      async (contractEntity): Promise<void> => {
+        await this.repositories.contractEntityRepository.save(contractEntity);
+      },
+    );
+
+    await Promise.all(saveContractEntityPromises);
+
+    await this.repositories.gatewayRepository.save(originGateway);
+    await this.repositories.gatewayRepository.save(auxiliaryGateway);
+
+    await this.repositories.anchorRepository.save(originAnchor);
+    await this.repositories.anchorRepository.save(auxiliaryAnchor);
```

Comment:
What will happen if any one of these 4 fails?
Should we use `sequelize.transaction`?  https://sequelize.org/v5/manual/transactions.html
Or
Make this as the last statement. `await this.repositories.gatewayRepository.save(originGateway);`
The seed data verifier checks only for the origin gateway address.

Reply:
Done, updated the sequence of repository call.